### PR TITLE
Document behavior of min/max/nanmin/nanmax for empty inputs

### LINKED ISSUE #3241: Document behavior of `min`, `max`, ... in case of empty, masked, ... inputs
See #3237, this really needs to be mentioned in the docs.

### PR DESCRIPTION
**Pull request by SimonHeybrock**
Fixes #3241.

### CODE REVIEW

**Review by jl-wynen — APPROVED** [review]

**Review by jokasimr — COMMENTED** [review]
> Scipp returns DBL_MAX for empty inputs, while NumPy returns NaN. For integer inputs, Scipp returns INT_MAX, while NumPy raises.

It is not really correct to say that Numpy returns Nan when the imput is empty, or at least it is a bit unclear. Both of the below examples raises `ValueError: zero-size array to reduction operation minimum which has no identity` independent of the data-type.

```python
np.min(np.ones(0, dtype=np.int64))
np.min(np.ones(0, dtype=np.float64))
```

However, `Numpy.min` does returns `nan` if the input contains `nan`.

----

As a side note that probably should be the topic of another issue: When looking into this issue I noticed that `sc.nanmin` and `sc.min` does the same thing

```python
# Both return -1
sc.array(dims=['x'], values=[1, float('nan'), -1]).min()
sc.array(dims=['x'], values=[1, float('nan'), -1]).nanmin()
```

while

```python
# Only nanmin returns -1
np.min([1, float('nan'), -1])
np.nanmin([1, float('nan'), -1])
```

I think the reason is that in the implementation of `sc.min` we use `std::min(a, b)` equivalent to `(b < a) ? b : a;` and if `b=nan` we get `a`.

**Comment by SimonHeybrock** [comment]
Hmm, and from `numpy.ma.masked_array` one gets a special value (of type `numpy.ma.core.MaskedConstant`):

```python
import numpy as np
import numpy.ma as ma

ma.masked_array(np.ones(1), mask=[True]).min()  # masked
```

File: src/scipp/core/reduction.py [hunk 1]
```diff
@@ -154,10 +154,10 @@ def min(x: VariableLikeType, dim: Optional[str] = None) -> VariableLikeType:
     Warning
     -------
 
-    Scipp returns DBL_MAX for empty inputs, while NumPy returns NaN. For integer
-    inputs, Scipp returns INT_MAX, while NumPy raises. Note that in the case of
-    :py:class:`DataArray`, inputs can also be "empty" if all elements contributing
-    to an output element are masked.
+    Scipp returns DBL_MAX and INT_MAX for empty inputs of float or int dtype,
```
Comment (jl-wynen) [inline src/scipp/core/reduction.py:157]:
```suggestion
    Scipp returns DBL_MAX or INT_MAX for empty inputs of float or int dtype,
```

File: src/scipp/core/reduction.py [hunk 2]
```diff
@@ -154,10 +154,10 @@ def min(x: VariableLikeType, dim: Optional[str] = None) -> VariableLikeType:
...
+    respectively, while NumPy rases. Note that in the case of :py:class:`DataArray`,
```
Comment (jl-wynen) [inline src/scipp/core/reduction.py:158]:
```suggestion
    respectively, while NumPy raises. Note that in the case of :py:class:`DataArray`,
```

File: src/scipp/core/reduction.py (hunk 1 above)
Comment (jl-wynen) [reply src/scipp/core/reduction.py:157]: And in the other docstrings, too.

**Review by jl-wynen — APPROVED** [review]

**Review by jokasimr — APPROVED** [review]

**Pull request merged**